### DISCUSSION TITLE
1.3.0 Beta started: Individual load preset for dual solarbank systems

### DISCUSSION THREAD

**Discussion by thomluther** [thomluther/ha-anker-solix#61] [Announcements]
## 1.3.0 Beta: Individual load preset for dual solarbank systems
If you have a dual solarbank setup and are interested to try the enhancements for individual device presets, you may install the beta release in the 1.3.0 branch.
The enhancements are included in the updated [INFO.md](https://github.com/thomluther/hacs-anker-solix/blob/1.3.0/INFO.md)
I have upgraded to this release as well with my single solarbank setup and haven't found any issues yet.
Beside the enhancements for dual solarbank systems, you may also try the other changes:

### Enhancements:
- Support for individual solarbank output power preset in dual solarbank systems:
  - New number entity to modify the device load preset
  - New sensor to indicate the active output power preset mode
  - Limits for appliance and device presets are now automatically adjusted based on number of installed solarbanks #48
  - Enhanced schedule services and added an optional device load preset, which also allows to specify the device load and appliance load in combination for the given interval. The difference will then be applied to the other solarbank automatically. This allows a single service call against one of the two solarbanks to adjust both individually
  - **Note: Individual device load preset requires solarbank firmware 1.5.6 or later**
  - The new entities will not be created for single solarbank systems or when the existing schedule structure does not support this feature yet.
  - After you updated both of your solarbanks to a supported firmware level but don't see the new entities yet, you need to reload the configuration of your Anker Solix integration to trigger a recreation of all entities.
- New entity to report on available OTA updates for solarbank devices
  - This binary sensor will flag when there is a firmware update available for your solarbank
  - Unfortunately the cloud api does not provide information about the available firmware version
  - The sensor can be used for notifications when an update is available, which has to be performed through the mobile app 
- Updated the [README](https://github.com/thomluther/hacs-anker-solix/blob/main/README.md) and [INFO](https://github.com/thomluther/hacs-anker-solix/blob/main/INFO.md) with the 1.3.0 enhancements

### Fixes:
- Missing preset entities when no schedule is defined bug dependencies #52 

## How to install beta with HACS
HACS provides an install update service which allows you to specify a specific version for the update. The version will first be looked up in the github repo tags (releases) and if not found there it will use the branch tags. So you can upgrade and downgrade your integration to any existing release or branch of the repo. Likewise you can also install the 'main' branch if that contains the latest code but has not been tagged with a release yet.

To install this beta from the 1.3.0 branch, just select the Anker-Solix update entity and specify the 1.3.0 version
![Install_update](https://github.com/thomluther/hacs-anker-solix/assets/97168515/dc36fb81-ed77-43b4-8128-3f6f888c0cea)

After installation, you need to restart home assistant.
If you find any problems or weird reporting, please enable debug mode of the integration, recreate the problem and open a new issue with the debug logs.

## Attention:
Since I cannot test the new version with a dual solarbank setup, I will leave this beta version active for a couple of weeks before releasing it as new version. Fixes or further improvements may require changes in the Api library too.

**Comment by Rilgamon** [comment]
Does it take a while before it shows up? Even after a restart of HA with enabled Beta-Switch I dont see it.

**Reply by thomluther** [reply]
This has nothing to do with Beta switch of HA core.
You need to install it manually with a service as shown in the screenshot

**Reply by Rilgamon** [reply]
Yes, but I can select the "Update: Update installieren" in services and there is no solix entitie in the list.
![grafik](https://github.com/thomluther/hacs-anker-solix/assets/471030/e09ffe5a-7105-48eb-b0ea-e15c62f58643)

**Reply by thomluther** [reply]
Hm, this is maybe related to how you have configured HACS.
I have it configured with expirimental features, because it then better integrates into core.
![image](https://github.com/thomluther/hacs-anker-solix/assets/97168515/eec372fc-7243-48b5-9e71-d6f135ccd9c3)
If you change it, you need to restart HA to have it active. Then you should see update entities for each HACS component.

Without the HACS experimental features, you should also be able to install different versions of the integration. Lookup the HACS anker solix tile, where you can also re-install the integration. I believe there is a switch to show also beta releases for an integration. With the legacy HACS features, you only have a selection list with versions found on the repo. You also may need to refresh the repo info for the integration in your HACS panel.

**Reply by Rilgamon** [reply]
Thanks, experimental features helped. Even without restart Solix showed up.

**Reply by thomluther** [reply]
see also here for generic [generic HACS update information](https://hacs.xyz/docs/basic/updates) and for [enabling HACS beta updates for an integration](https://hacs.xyz/docs/faq/select_version)
However, I'm not sure if branches on github count as beta release and will show up

**Reply by Rilgamon** [reply]
1.3.0 is installed now :)

**Reply by Rilgamon** [reply]
Settings work well. Lets see what happens when the sun is available.

**Comment by thomluther** [comment]
Some topics to be tested:
- Are new entities created when no schedule is defined during integration (re)load
- Does the Set Service work with various combinations of appliance and device load parameters (single and both)
  - does this work when no schedule was defined
- Does the update service work with various combinations of appliance and device load parameters (single and both)
  - Are existing interval settings re-used when either start or end time are the same
  - Does this re-use work properly for various combinations of appliance and device load parameters (single and both, or none)
- Are the provided settings really applied and the home load is delivered as expected

**Attention:**
It was observed that sending a single solarbank schedule time interval via the cloud Api may result in a weird schedule on the solarbank. While the cloud and the integration shows correct output preset values, the appliance has a 0W preset. This cannot be corrected via the cloud Api. Any interval update or addition will result in the 0 W preset for all intervals. This will cause that the solarbank does NOT discharge anymore although the settings show correctly in the cloud and the HA integration. This firmware issue may depend on your firmware and your inverter configuration vs. real inverter setup. I noticed this problem while having configured MI80 inverter (to allow min. of 100 W preset), but not having a MI80 really installed. You can correct this problem only via the Anker mobile app which seems to use a different process to modify and query the solarbank schedule than via cloud Api.
I still ran into this issue with 1.6.2 firmware. So setting just a single interval in total with a schedule update call may result in the inverter being switched off, even if the interval is supposed to deliver output.

**Comment by Rilgamon** [comment]
First test with disabled inverter limit (1600W) shows peak output at 1500W with really slow adjustments.  Did not see this slow changes when used in my normal range 400-600W. Reducing only one from 800 to 260 quickly reduces to less than required and climbs back up slow.
![grafik](https://github.com/thomluther/hacs-anker-solix/assets/471030/7715ebc5-9287-4566-b3da-ec16ffc2775b)

**Reply by thomluther** [reply]
That seems to be the way the solarbank is regulating the discharge. What I have read on Solarbank 2, this should regulate faster, which is also required for 0 W grid injection.
As long as the values show correctly in the integration and the final result is as expected, the integration works good and it shows that you can now go beyond 800 W and the appliance also reacts on it.

When allowing more than 600/800 W output for longer periods, you should make sure that no other consumers are on the circuit or you have a separate inverter circuit with its own circuit breaker to avoid risk of overloading cables.

**Reply by thomluther** [reply]
Maybe I should note, that changing any entity of the schedule interval will directly result in a schedule change and the solarbank will react on it.
So if you first change the device load, then the appliance load or vice versa, each change is sent immediately.
If you want to change both entities at the same time, you have to use the update schedule service with the current interval. Then the resulting change of appliance load and individual device load changes are done in a single api request

**Reply by Rilgamon** [reply]
No, just disabled it for this test. Because I asked for it when I ran into the "problem" with my dynamic adjustment. Overall it works really well. Switching back and forth from normal mode, setting one or the other device or back to system load. Love it!
With your integration I think its the better choice for smart home to use the "old" solix. 
Hope to see some sun this/next week to test (without) schedules. Atm there is a constant flow of weather warnings coming in ;)

**Reply by thomluther** [reply]
Modification without schedule is not supported. The appliance will allways have fixed export of 200 W if it have no interval defined for the actual time (this is also written in the mobile app). 
I don't know if 200 W export applies for dual systems as well, it but would be good to test that each device only exports 100 W when no schedule is set. This is my assumption when I report preset sensors in case no schedule is active for current time.

If you modify a schedule entity with the integration, it will make sure that an all day interval is defined to apply the changed value and use interval defaults for the other values. Otherwise without any schedule (or current gap in the schedule), the entity change could not be applied. This is what makes the api integration so complex, because you need to send always a schedule interval with all parameters.

**Comment by jonashein** [comment]
I'm also running 1.3.0 now. I did some tests both on the python API and the HA integration, but haven't found any issues yet. If I have some spare time I'll try to specifically test the scenarios you mentioned [above](https://github.com/thomluther/hacs-anker-solix/discussions/61#discussioncomment-9549225).

I just integrated the new feature into my net-zero power script, and so far it works well (if there's not too many clouds passing by ;) )
![image](https://github.com/thomluther/hacs-anker-solix/assets/16691259/40b619e5-78a9-4393-b6fa-202eb02357fe)

Thanks for your efforts!

**Comment by thomluther** [comment]
Question to dual solarbank system owners:
When you look at the energy diagrams in the Anker app, can you see different statistics for each solarbank?
I assume the solar diagramm and statistics should be the same for both solarbanks, but when you look at the battery charge and discharge, is that separate per device or only available for the appliance?
Currently the Api queries them per device, but if the stats are the same and on appliance level only, the energy stats queries could be reduced for dual solarbank systems and numbers simply re-used per device in the system.

**Reply by jonashein** [reply]
There's only the one solar production and solarbank storage diagram for the whole system, but no option to display per-solarbank statistics.

**Reply by thomluther** [reply]
Ok thanks. How do the daily numbers match with the energy statistics you get in HA per Solarbank, and are the 2 Solarbank values identical for daily solar energy, charge and discharge energy?
I need to understand whether the Api delivers only appliance values independently of the queried device SN, or whether the App diagram just accumulates the individual device energies.

**Reply by jonashein** [reply]
Both solarbanks show the same values for daily charge, daily discharge and daily solar yield, and these values match with the values shown in the app statistics.

Summing up the power integrals in HA I get somewhat similar values to the ones above, so it seems the statistics are for the system.

**Reply by thomluther** [reply]
Ok thank you for confirmation. That means the energy related entities should rather be tracked with the system instead of the device.
This will also save api queries when only one solarbank per system has to be queried.
But it will be a disruptive change moving them to the system...need to think about that for a future release. Solarbank 2 and Smartreader come with additional energy statistics for the system, so this will have to be changed anyway
